gapic/schema/api.py

```diff
@@ -221,16 +236,43 @@ def disambiguate_keyword_fname(
 
         # Iterate over each FileDescriptorProto and fill out a Proto
         # object describing it, and save these to the instance.
-        protos: Dict[str, Proto] = {}
+        #
+        # The first pass gathers messages and enums but NOT services or methods.
+        # This is a workaround for a limitation in protobuf annotations for
+        # long running operations: the annotations are strings that reference
+        # message types but do not require a proto import.
+        # This hack attempts to address a common case where API authors,
+        # not wishing to generate an 'unused import' warning,
+        # don't import the proto file defining the real response or metadata
+        # type into the proto file that defines an LRO.
+        # We just load all the APIs types first and then
+        # load the services and methods with the full scope of types.
+        pre_protos: Dict[str, Proto] = dict(prior_protos or {})
         for fd in file_descriptors:
-            fd.name = disambiguate_keyword_fname(fd.name, protos)
-            protos[fd.name] = _ProtoBuilder(
+            fd.name = disambiguate_keyword_fname(fd.name, pre_protos)
+            pre_protos[fd.name] = Proto.build(
```

Comment:
Is it possible to refactor `Proto.build` so that it loads all messages and enums first, and then build methods and services? So instead of calling `build` twice, you could have build call part of itself to "prebuild". It seems cleaner. I don't know the implementation details so this might be impossible or very hard in which case ignore me :)

Reply:
There are a couple of tricky issues to work around:
* `Proto` is a frozen datatype, so once it's been created it is immutable.
* We need some kind of two stage solution so that we load _all_ the proto messages, and then methods can see all message and enum types.

I would certainly be willing to rework the current implementation so that messages and enums are loaded and wrapped in rich structures, then services and methods are loaded and wrapped in rich structures using the messages and enums, then `Proto` is created by injecting messages, enums, services, and methods. This would require a larger change, but would be cleaner and more aesthetically pleasing. I could go either way.

Reply:
Sounds like a lot of work :( You don't have to block on this comment. Feel free to merge if the fix is pressing and the Python developers sign off.

Reply:
I definitely want to rework the internal implementation, but that is going to take some time. I'll file a cleanup issue around it but commit the current impl as is, pending approval.

Reply:
Tracked by #331